src/app/data-planes/components/BuiltinGatewayPolicies.vue

```diff
@@ -17,7 +17,7 @@
               name: 'policy-detail-view',
               params: {
                 mesh: policy.mesh,
-                policyPath: props.policyTypesByName[policy.type]!.path,
+                policyPath: props.types[policy.type]?.path ?? '',
```

Comment:
**Question**: Why is this being changed? This doesn’t look right. We should assume that a policy type appearing anywhere has a corresponding object found on the `/policies` API response. Allowing for `''` as a policy path is never correct so this code shouldn’t look like that’s possible. (Same with similar occurrences in this change set)

Reply:
I was looking to "deal with it" rather than "ignore it". 

> Allowing for '' as a policy path is never correct so this code shouldn’t look like that’s possible.

Allowing for an `undefined` access error is also never correct. Thing is with `''` we can probably deal with it inside our own `XAction` and fail a little more gracefully than just crashing.

I kinda don't mind either way right now seeing as we aren't using `XAction` everywhere yet. To be fair at the moment the error we get is with the `undefined` access is more descriptive should we ever get an error (highly unlikely I think?), but that reason will go away once we fail more gracefully.

Let me know if you feel strongly about this and I'd be happy to change it back.

Was there anything else in this PR or was it just this?

Reply:
@kleinfreund just noticed this is still outstanding, any thoughts on this?

Reply:
I guess I don’t agree with the philosophy behind why this is better. If you ever get to the `''` code path, something’s already gone wrong before this point and the user should see an error of some kind.

What we’re dealing with here is basically that we aren’t able to model that `props.policyTypesByName[policy.type]` is a safe access on the type level. Solving this with a type assertion seems appropriate because it keeps the decision where the problem arises: on the type level. Using a conditional property access with a fallback value is more intransparent and signals "hey, if there isn’t a fitting object, fall back to this value". But that’s a nonsensical application scenario.

In any case, I don’t care too strongly about it as both variants work. Resolving this.